react/utils/pages.ts

```diff
@@ -154,29 +154,36 @@ export function navigate(
   const {
     page,
     params,
-    query,
-    to,
+    query: inputQuery,
+    to: inputTo = '',
     scrollOptions,
     fallbackToWindowLocation = true,
     rootPath,
     replace,
     fetchPage = true,
   } = options
 
-  if (!page && !to) {
+  if (!page && !inputTo) {
     console.error(
       `Invalid navigation options. You should use 'page' or 'to' parameters`
     )
     return false
   }
+  
+  const [to, extractedQuery] = inputTo.split('?')
+  const query = inputQuery || extractedQuery
+
+  if (inputTo && inputQuery) {
+    console.warn(`You shouldn't pass 'query' in a separate prop when using 'to'`)
```

Comment:
actually, why not? I think there should be a warning only if the user is passing both a `query` and a query string on `to`. otherwise, there's no harm, is there?

Reply:
I've discussed with @brunoabreu about this and it's more concise defining only one way to pass queries. Since when using _page_, there's no way to pass the queries there, you need another way, but when using _to_, everything should be passed there.

We are still supporting, but not encouraging

Reply:
@lbebber isn't exatly that what this if statement is doing? :thinking:

Reply:
ah, I misread it. carry on 👌